lib/ui/common/Alert.dart

```diff
@@ -0,0 +1,223 @@
+import 'package:farmsmart_flutter/ui/common/roundedButton.dart';
+import 'package:farmsmart_flutter/ui/mockData/MockRoundedButtonViewModel.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/foundation.dart';
+
+class _Constants {
+  static final EdgeInsets alertEdgePadding =
+      const EdgeInsets.symmetric(horizontal: 24);
+  static final BorderRadius cornerRadius =
+      const BorderRadius.all(Radius.circular(24.0));
+  static final EdgeInsets alertInnerPadding =
+      const EdgeInsets.only(left: 32, right: 32, bottom: 31, top: 31);
+  static final double titleLineSpace = 19;
+  static final double detailLineSpace = 28;
+  static final double actionHeight = 48;
+  static final double actionWidth = 120;
+  static final BorderRadius actionCornerRadius =
+      const BorderRadius.all(Radius.circular(14));
+  static final double actionLineSpace = 8;
+}
+
+class AlertViewModel {
+  String cancelActionText;
+  String confirmActionText;
+  String titleText;
+  String detailText;
+  Function confirmAction;
+  bool isDestructive;
+
+  AlertViewModel({
+    @required this.cancelActionText,
+    @required this.confirmActionText,
+    @required this.titleText,
+    this.detailText,
+    this.confirmAction,
+    this.isDestructive,
+  });
+}
+
+class AlertStyle {
+  final Color backgroundColor;
+  final TextStyle titleTextStyle;
+  final TextStyle detailTextStyle;
+  final Color actionBackgroundColor;
+  final Color destructiveActionBackgroundColor;
+  final TextStyle actionTextStyle;
+
+  const AlertStyle({
+    this.backgroundColor,
+    this.titleTextStyle,
+    this.detailTextStyle,
+    this.actionBackgroundColor,
+    this.destructiveActionBackgroundColor,
+    this.actionTextStyle,
+  });
+
+  AlertStyle copyWith({
+    Color backgroundColor,
+    TextStyle titleTextStyle,
+    TextStyle detailTextStyle,
+    Color actionBackgroundColor,
+    Color destructiveActionBackgroundColor,
+    TextStyle actionTextStyle,
+  }) {
+    return AlertStyle(
+      backgroundColor: backgroundColor ?? this.backgroundColor,
+      titleTextStyle: titleTextStyle ?? this.titleTextStyle,
+      detailTextStyle: detailTextStyle ?? this.detailTextStyle,
+      actionBackgroundColor:
+          actionBackgroundColor ?? this.actionBackgroundColor,
+      destructiveActionBackgroundColor: destructiveActionBackgroundColor ??
+          this.destructiveActionBackgroundColor,
+      actionTextStyle: actionTextStyle ?? this.actionTextStyle,
+    );
+  }
+}
+
+class _DefaultStyle extends AlertStyle {
+  final backgroundColor = const Color(0xffffffff);
+  final titleTextStyle = const TextStyle(
+    color: Color(0xff1a1b46),
+    fontSize: 27,
+    fontWeight: FontWeight.bold,
+  );
+  final detailTextStyle = const TextStyle(
+    color: Color(0xff1a1b46),
+    fontSize: 17,
+    height: 1.1,
```

Comment:
why do we set the height?

Reply:
height is in TextStyle are the LineSpace separation, so we need to make some request changes to the flutter guys to correct that naming xD

Reply:
poor naming google :) should be linespacing